From d0e6880f0111b5929c8ca37c542d90075c95768c Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 25 Apr 2024 09:15:26 +0200
Subject: [PATCH 0837/1695] core: pta: HWRNG enumerate upon registration

Allow HWRNG PTA to request be enumerated only when a TRNG is registered.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git aedde51088a9b6d34ecefdfb0de6e53b6fd9bf50 core: pta: HWRNG enumerate upon registration]
Change-Id: Ia74c1ae0226ff774ee0ec22d2822ef6639a52fad
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/377166
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/pta/hwrng.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/pta/hwrng.c b/core/pta/hwrng.c
index ed089a10b..25a9413ad 100644
--- a/core/pta/hwrng.c
+++ b/core/pta/hwrng.c
@@ -74,7 +74,13 @@ static TEE_Result invoke_command(void *session __unused,
 	return TEE_ERROR_NOT_IMPLEMENTED;
 }
 
+static bool do_enumerate(void)
+{
+	return crypto_rng_hw_is_registered();
+}
+
 pseudo_ta_register(.uuid = PTA_RNG_UUID, .name = PTA_NAME,
 		   .flags = PTA_DEFAULT_FLAGS | TA_FLAG_CONCURRENT |
 			    TA_FLAG_DEVICE_ENUM,
-		   .invoke_command_entry_point = invoke_command);
+		   .invoke_command_entry_point = invoke_command,
+		   .ask_for_enumeration = do_enumerate);
-- 
2.47.2

